RecoMuon/DetLayers/src/MuonME0DetLayerGeometryBuilder.cc

```diff
@@ -103,15 +106,17 @@ MuonME0DetLayerGeometryBuilder::buildLayer(int endcap,
     }
 
   }
-  LogTrace(metname) << "About to make a MuRingForwardLayer";
-  result = new MuRingForwardLayer(frontRings);
-  
-  LogTrace(metname) << "New MuRingForwardLayer with " << frontRings.size()
-                    << " and " << backRings.size()
-                    << " rings, at Z " << result->position().z()
-                    << " R1: " << result->specificSurface().innerRadius()
-                    << " R2: " << result->specificSurface().outerRadius();
   
+  LogTrace(metname) << "About to make a MuRingForwardLayer";
+  if(frontRings.size()!=0) result = new MuRingForwardLayer(frontRings);
```

Comment:
@cvuosalo - looking at this further, these are already assigned as std::shared_ptr upstream in https://github.com/cms-sw/cmssw/blob/CMSSW_8_1_X/RecoMuon/DetLayers/plugins/MuonDetLayerGeometryESProducer.cc#L98